scantool/scantool_850.c

```diff
@@ -693,6 +701,34 @@ static void interpret_value(enum l7_namespace ns, uint16_t addr, UNUSED(int len)
 	} else if (ns==NS_LIVEDATA && addr==0x1000) {
 		/* ECU pin A4, MCU P7.4 input, divider ratio 8250/9460 */
 		printf("MAF sensor signal: %.2f V\n", (float)buf[0]*9460/8250*5/256);
+	} else if (ns==NS_LIVEDATA && addr==0x1800) {
+		printf("Short term fuel trim: %+.1f%%\n", (float)buf[0]*100/128-100);
+	} else if (ns==NS_LIVEDATA && addr==0x1900) {
+		/* possibly in units of 0.004 milliseconds (injection time) */
+		printf("Long term fuel trim, additive (unscaled): %+d\n", (signed int)buf[0]-128);
+	} else if (ns==NS_LIVEDATA && addr==0x1A00) {
+		printf("Long term fuel trim, multiplicative: %+.1f%%\n", (float)buf[0]*100/128-100);
+	/* ECU 6E: */
+	} else if (ns==NS_LIVEDATA && addr==0x0500) {
+#if 0
+		/* Bench tested, but not verified on vehicle */
+		printf("Mode selector: MS1 %s, MS2 %s, switch position %s\n", (buf[0]&1)?"low":"high", (buf[0]&2)?"low":"high", mode_selector_positions[(buf[0]>3)?4:buf[0]]);
+		printf("Driving mode: %s\n", driving_modes[(buf[1]>2)?3:buf[1]]);
+#endif
+	} else if (ns==NS_LIVEDATA && addr==0x0C00) {
+		/* Voltage scale factor looks like 2.04 in bench testing,
+		   but 2.048 makes more sense. */
+		volts = ((float)buf[0]*256+buf[1])*10/2048;
+		printf("ATF temperature sensor voltage: %.2f V\n", volts);
+		/* Input has 1k to +5V, sensor acts as a potential divider */
+		if (volts >= 5) {
+			printf("ATF temperature sensor resistance: open circuit\n");
+		} else {
+			printf("ATF temperature sensor resistance: %u ohms\n", (uint16_t)((1000.0f*volts)/(5.0f-volts)));
```

Comment:
why the cast to uint16 instead of just unsigned ?

Reply:
Changed to unsigned.